server/server.js

```diff
@@ -110,12 +136,33 @@ var upload = multer({
  */
 var exonFile = dataPath + 'exons.bin';
 
-app.use(bodyParser.urlencoded({extended: true}));
-app.use(bodyParser.json());
+/**
+ * User authenticate handler.
+ */
+MongoClient.connect(mongoUrl, function(err, db) {
+  if (err) {
+    log.serverLog(err.message);
+    return;
+  }
+  log.serverLog('connected to MongoDB');
+  database.db = db;
+
+  // Start the application.
+  //var privateKey = fs.readFileSync(privateKeyPath);
+  //var certificate = fs.readFileSync(certificatePath);
+  //var httpsServer = https.createServer({
+  //  key: privateKey,
+  //  cert: certificate
+  //}, app).listen(443);
+  //httpsServer.setTimeout(1200000);
+
+  var server = app.listen(3000);
+  server.setTimeout(1200000);
+});
 
 var serverResponse = function(data, res) {
```

Comment:
When should it return 200? Success or fail?

On Fri, Apr 1, 2016 at 2:39 PM, Bowen Yu <notifications@github.com> wrote:

> In server/server.js
> <https://github.com/ViDA-NYU/genotet/pull/145#discussion_r58248829>:
>
> > +  }
> > +  log.serverLog('connected to MongoDB');
> > +  database.db = db;
> > +
> > +  // Start the application.
> > +  //var privateKey = fs.readFileSync(privateKeyPath);
> > +  //var certificate = fs.readFileSync(certificatePath);
> > +  //var httpsServer = https.createServer({
> > +  //  key: privateKey,
> > +  //  cert: certificate
> > +  //}, app).listen(443);
> > +  //httpsServer.setTimeout(1200000);
> > +
> > +  var server = app.listen(3000);
> > +  server.setTimeout(1200000);
> > +});
> >
> >  var serverResponse = function(data, res) {
>
> Okay. Then it is not necessary to res.json(data). You can just reply with
> a 200 header.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/ViDA-NYU/genotet/pull/145/files/314834a4206e43d9be8af643b14ca22ccb2962e1#r58248829>
>